amt.go

```diff
@@ -500,6 +500,7 @@ func (n *Node) Flush(ctx context.Context, bs cbor.IpldStore, depth int) error {
 		if len(n.expVals) == 0 {
 			return nil
 		}
+	  n.Bmap = [...]byte{0}
```

Comment:
```suggestion
		n.Bmap = [...]byte{0}
```

Reply:
We should probably set this once at the top of the function. The return on line 517 could be wrong (no links but a non-empty bitfield).

Reply:
Probably not safe to do so, there's a possibility of dealing with a node that has intact `Values` or `Links` and a correct `Bmap` but not the expanded form in `expVals` or `expLinks`. Setting the `Bamp` to `0` will put the node in an inconsistent state and if continue down to a rebuild we'll have an empty Values or Links array.

Reply:
Ah, I see. In that case, I'm not sure _why_ we even rebuild this. From what I can tell, we should keep it up-to-date on every operation.

Reply:
Yes, and in doing that there's an option to do away with `expLinks` and `expValues` entirely and just keep more careful track of index translation through `Bmap`. They do add a lot of convenience for some operations but it wouldn't be hard to add in convenience functions to help with indexing. I don't think they add much in the way of performance, they could even hinder performance because they're used so liberally (e.g. a `get()` should really just map index through `Bmap` and fetch directly out of `Values` or `Links`, no need to expand just to read). And you have these extra structures hanging around in memory that aren't necessary so they'll add a little to memory bloat.